README.md

```diff
@@ -89,6 +89,35 @@ The above command will create a receiver named ```Librespot```, with bitrate set
 A full list of runtime options are available [here](https://github.com/librespot-org/librespot/wiki/Options)
 
 _Please Note: When using the cache feature, an authentication blob is stored for your account in the cache directory. For security purposes, we recommend that you set directory permissions on the cache directory to `700`._
+
+### Running as a system-wide service
+
+`contrib/systemd.service` should be installed into `/usr/lib/systemd/system/`.
+This'll allow running librespot as a system-wide service permanently by
+enabling it:
+
+    systemctl enable --now librespot
+
+This is the recommended approach for embedded devices. Note that only `root`
+can manage this service.
+
+It is recommended that downstream packagers include this file in distribution
+packages.
+
+### Running as a user service
+
+`contrib/systemd.user.service` should be installed into `/usr/lib/systemd/user/`.
+This'll allow running librespot as a user service which can be controlled
+without privileges, and is bound to the currently logged in user.
+
+    systemctl --user enable --now librespot
+
+This is the recommended approach for desktops and especially for shared devices
```

Comment:
Is it?

Reply:
I don't think `librespot` includes another way to run as a service in this scenario.

I can remove this line (and the same above) if you disagree on this item.

Reply:
Because most of the time a system service is totally fine for us. The idea that each user needs to run their own copy or somehow manage the service is not relevant. It's only useful in terms of avoiding extra pulseaudio config otherwise i'd say don't even bother including it at all.

Reply:
> Because most of the time a system service is totally fine for us. The idea that each user needs to run their own copy or somehow manage the service is not relevant. It's only useful in terms of avoiding extra pulseaudio config otherwise i'd say don't even bother including it at all.

It is if your system runs PulseAudio or possibly PipeWire. PulseAudio runs as a user level service. If you run librespot on a desktop system with PulseAudio the easiest way is a user service.

Reply:
Which is pretty much what I said:

> it's only useful in terms of avoiding extra pulseaudio config

Reply:
Avoiding mucking with and potentially borking PulseAudio on your system by following some sketchy online tutorial vs a stupid easy user service. That seems like a really good trade off...

Reply:
I agree with that much.

Reply:
In fact, that should be explicitly be stated. It's the main reason to use this and it's not mentioned anywhere.

Reply:
> In fact, that should be explicitly be stated. It's the main reason to use this and it's not mentioned anywhere.

The other use case is a multi user system like a family computer. Dad and little Johnny might want the service enabled when they're logged in but Mom and little Susie don't. That's implied but not really stated out right. The permissions thing is stressed and I feel like users more about making it do what they want it to then permissions. But I would agree that PulseAudio compatibility is at least an equal draw if not more so. And I also agree that the explicit intent and reason should be stated otherwise you just end up with what we have here. A bunch of people wondering why.

Reply:
I also don't agree with the idea that downstream should include these files copypasta style. They could be used as examples for diy types sure but any packager should know how to write a unit file.

Reply:
I also kinda feel like this belongs in the wiki not the README. There's are ready a [Running-in-the-background](https://github.com/librespot-org/librespot/wiki/Running-in-the-background) section in the wiki. This could be added to that.

Reply:
I'll add it to the wiki and remove the reference from the README, that seems to be the best place for now.

Note that, aside from being tied to a user session, it's also controllable by the user. The system service cannot be started/stopped/reconfigured by unprivileged users.

Having downstream write the service files results in a lot of duplicate work, and frequently tiny, confusing, differences across distributions.

Reply:
> Having downstream write the service files results in a lot of duplicate work, and frequently tiny, confusing, differences across distributions.

Better duplicate work than copypasta. One would hope that those tiny confusing differences are for a reason. Different distros different use cases and philosophies. I just don't want `librespot` saddled with dealing with bugs that downstream should be dealing with. `librespot` isn't really about creating a ready to go systemd service daemon. It's a collection of libs that just so happen to be able to compiled together to form a functioning Spotify Connect client and that great I love that, but figuring out how to use that binary is a job for downstream.

Reply:
I don't quite follow. What copypasta are you talking about?

Downstream packagers won't manually copy-paste the file, package generations are scripted. The file will be copied from the repository.

Reply:
I've moved this documentation to the wiki.

Reply:
If it's copied directly from the repo it's basically copypasta. What I mean is that packagers should think about their specific use case and write their own unit file. Starting with the services in the repo as an example is fine but they shouldn't just be automatically used without thought or consideration.